### PR TITLE
Update contribution guide on how to bump a task/pipeline

### PR DESCRIPTION
**Pull request by chmouel**
Update contributiong guide with guidance on  how to bump a task and pipeline
make it easy for reviewers and have a clean git log.

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
/cc @tektoncd/catalog-collaborators 

let's maybe set this as policy

**Review by vdemeester — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,6 +28,31 @@ The process for contributing looks like this:
...
+
+```shell
+% git add tasks/foo/0.2
+% git commit -m "Copy task foo from 0.1 to 0.2
+```
+
+and then add your change and commit it.
+
+This will result to a clean git log and makes it easier to only see your
+changes.
```
Comment (vinamra28) [inline CONTRIBUTING.md:54]: one question over here :sweat_smile: 
should we ask the author of PR to squash the commits before merging or is it fine to merge as it is?

**Comment by sbwsg** [comment]
I'm a big fan of this change. I don't feel too strongly about whether we require squash or not.

One observation though - there is a vector here by which a malicious user could introduce code that doesn't get properly reviewed. By labelling one of the commits a simple copy as a reviewer I find myself "trusting" that commit without reading it fully. A malicious dev could introduce changes in that commit and it might not get the same level of scrutiny as it did when the whole thing was being reviewed in detail.  I think this is _also_ a problem when the changes are mixed up with the copy operation in a single commit but I guess I just wanted to point it out.

I'm happy to lgtm but will wait til the question from @vinamra28 is resolved.

**Comment by chmouel** [comment]
> One observation though - there is a vector here by which a malicious user could introduce code that doesn't get properly reviewed. By labelling one of the commits a simple copy as a reviewer I find myself "trusting" that commit without reading it fully. A malicious dev could introduce changes in that commit and it might not get the same level of scrutiny as it did when the whole thing was being reviewed in detail. I think this is also a problem when the changes are mixed up with the copy operation in a single commit but I guess I just wanted to point it out.

@sbwsg You are right but I think the alternative of having reviewer the full task and potentially miss the changes may be as problematic.

If someone has time it would be great if we can just integrate this : 

https://github.com/tektoncd/catalog/blob/main/hack/diff-task-version

on every PR iteration, making sure nothing "slips in". I think there is maybe a few fixes to do here in there in that script too.

this would work in conjunction of having a clean git commit log for tracking as described here.

**Comment by chmouel** [comment]
@vinamra28 imho we don't have to have a hard policy and it should be up to the reviewer appreciation.

I mean, when the changes are silly commits like thoses :

- fix typos.
- fix last commit failure
- address reviewer comment

then for sure I would kindly ask for a  rebase

but if the user add multiple "features" or pretty obvious fix and cared enough about splitting in different commits i.e: 

- add ability to specify a new user.
- add support for optional secrets workspace

then we should let them go.

moreover, i find that some newbies dev/contributors may not be very confident with the git rebase command, so we may have to get ready to help and guide them if we ask for a rebase

**Comment by vinamra28** [comment]
> > One observation though - there is a vector here by which a malicious user could introduce code that doesn't get properly reviewed. By labelling one of the commits a simple copy as a reviewer I find myself "trusting" that commit without reading it fully. A malicious dev could introduce changes in that commit and it might not get the same level of scrutiny as it did when the whole thing was being reviewed in detail. I think this is also a problem when the changes are mixed up with the copy operation in a single commit but I guess I just wanted to point it out.
> 
> @sbwsg You are right but I think the alternative of having reviewer the full task and potentially miss the changes may be as problematic.
> 
> If someone has time it would be great if we can just integrate this :
> 
> https://github.com/tektoncd/catalog/blob/main/hack/diff-task-version
> 
> on every PR iteration, making sure nothing "slips in". I think there is maybe a few fixes to do here in there in that script too.
> 
> this would work in conjunction of having a clean git commit log for tracking as described here.

@frerikandriessen is working on adding it as a tekton job.

**Comment by vinamra28** [comment]
> @vinamra28 imho we don't have to have a hard policy and it should be up to the reviewer appreciation.
> 
> I mean, when the changes are silly commits like thoses :
> 
> * fix typos.
> * fix last commit failure
> * address reviewer comment
> 
> then for sure I would kindly ask for a rebase
> 
> but if the user add multiple "features" or pretty obvious fix and cared enough about splitting in different commits i.e:
> 
> * add ability to specify a new user.
> * add support for optional secrets workspace
> 
> then we should let them go.
> 
> moreover, i find that some newbies dev/contributors may not be very confident with the git rebase command, so we may have to get ready to help and guide them if we ask for a rebase

yeah I agree with you. Okay let's keep this easy.
/lgtm

**Pull request merged**